unit/java_bytecode/java_bytecode_parse_generics/parse_nested_generics.cpp

```diff
@@ -0,0 +1,1339 @@
+/*******************************************************************\
+
+ Module: Unit tests for parsing generic classes
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+#include <testing-utils/require_symbol.h>
+#include <testing-utils/require_type.h>
+
+#include <util/cmdline.h>
+#include <util/config.h>
+#include <util/language.h>
+#include <util/prefix.h>
+#include <util/std_types.h>
+
+#include <java_bytecode/java_bytecode_language.h>
+
+#include <iostream>
+#include <testing-utils/load_java_class.h>
+
+SCENARIO(
+  "parse_nested_generics_fields",
+  "[core][java_bytecode][java_bytecode_parse_generics]")
+{
+  const symbol_tablet &new_symbol_table = load_java_class(
+    "NestedGenerics",
+    "./java_bytecode/java_bytecode_parse_generics");
+
+  std::string class_prefix = "java::NestedGenerics";
+  THEN("There should be a symbol for NestedGenerics with correct components")
+  {
+    REQUIRE(new_symbol_table.has_symbol(class_prefix));
+    const symbolt &class_symbol = new_symbol_table.lookup_ref(class_prefix);
+
+    const class_typet &class_type =
+      require_symbol::require_complete_class(class_symbol);
+
+    THEN("The field component should be a pointer to java::Generic")
+    {
+      const struct_typet::componentt &field_component =
+        require_type::require_component(class_type, "field");
+
+      require_type::require_pointer(
+        field_component.type(),
+        symbol_typet("java::Generic"));
+
+      THEN("The pointer should be generic")
+      {
+        REQUIRE(is_java_generic_type(field_component.type()));
+        const auto &generic_variables =
+          to_java_generic_type(field_component.type()).generic_type_variables();
+        REQUIRE(generic_variables.size() == 1);
+        {
+          const java_generic_parametert &generic_param = generic_variables[0];
+          REQUIRE(is_java_generic_parameter(generic_param));
+          REQUIRE(generic_param ==
+                  java_generic_inst_parametert(symbol_typet("java::Generic")));
+        }
+        // TODO: extend tests when nested generics are parsed correctly
```

Comment:
Jira reference?

Reply:
I think this is [TG-610] (but I agree would be good to add into the TODO)